### PR TITLE
Add envs for artifacts, toolset, tools, and log directories

### LINKED ISSUE #2276: Arcade should create environment variables for Artifacts and Logging directories
This should act like `Build.SourcesDirectory` where Arcade produces something like `Build.OutputDirectory` and `Build.LoggingDirectory`

### PR DESCRIPTION
**Pull request by zsd4yr**
fixes #2276

### CODE REVIEW

**Comment by zsd4yr** [comment]
> Let's not use environment variables. I though we agreed on using VSO variables.

I thought you said:

> @zsd4yr Feel free to send a PR that sets this variables at https://github.com/dotnet/arcade/blob/master/eng/common/tools.ps1#L522 ?? I am just following the pattern here

**Comment by tmat** [comment]
I also said:

> We set some variables to control dontet cli here: 
> https://github.com/dotnet/arcade/blob/master/eng/common/tools.ps1#L147
> We could set `Artifacts.Bin` and `Artifacts.Log` variables in the same way here: 
> https://github.com/dotnet/arcade/blob/master/eng/common/tools.ps1#L522

> I am just following the pattern here

The `TMP`/`TEMP` variables have different purpose. They are set there to redirect `GetTempPath`.

If you set environment variables they will be only available in the child processes. They won't be visible to the subsequent steps of the build pipeline.

File: eng/common/tools.ps1 [hunk 1]
```diff
@@ -518,6 +518,10 @@ Create-Directory $TempDir
 Create-Directory $LogDir
 
 if ($ci) {
+  Write-Host "##vso[task.setvariable variable=Artifacts] $ArtifactsDir"
```
Comment (tmat) [inline eng/common/tools.ps1]: Not sure if the ` ` in between `]` and `$` is significant and will become part of the variable value. It'd be probably better to remove it.

Comment (zsd4yr) [reply eng/common/tools.ps1]: happy to; I wasn't sure either tbh; a little easier to read this way if it works tho hehe

**Comment by markwilkie** [comment]
thanks @zsd4yr !

**Comment by zsd4yr** [comment]
> thanks @zsd4yr !

thanks guys for working with me on this :) and thanks to @tmat for guidance and clarification

**Comment by tmat** [comment]
Glad to help.

**Comment by zsd4yr** [comment]
@tmat looping back around here; these variables to no appear to resolve in Helix -- any suggestions? 

here's an example: https://dev.azure.com/dnceng/public/_build/results?buildId=140672

**Comment by markwilkie** [comment]
@alexperovich - any suggestions for passing these vars into Helix?  

cc/ @jonfortescue

**Comment by alexperovich** [comment]
That build is trying to access the variable $(LogDir), the logging commands use the variable $(Artifacts.Log)

**Comment by zsd4yr** [comment]
> That build is trying to access the variable $(LogDir), the logging commands use the variable $(Artifacts.Log)

😢 got me!

**Pull request merged**